stdpopsim/arabidopsis_thaliana.py

```diff
@@ -0,0 +1,125 @@
+"""
+Genome, genetic map and demographic model definitions for Arabidopsis thaliana.
+"""
+import math
+
+import msprime
+
+import stdpopsim.models as models
+import stdpopsim.genomes as genomes
+import stdpopsim.genetic_maps as genetic_maps
+
+
+###########################################################
+#
+# Genetic maps
+#
+###########################################################
+
+
+class Salome2012(genetic_maps.GeneticMap):
+    """
+    Genetic map from Salome 2012 averaged across population crosses.
+    Please see README for details on how this was done.
+    TODO: make README file
+
+
+    """
+    url = ("https://github.com/LohmuellerLab/stdpopsim/raw/master/"
+           "salome2012_maps.tar.gz")
+    file_pattern = "arab_{name}_map_loess.txt"
+#	TODO: need to find a place to host recombination maps
+
+genetic_maps.register_genetic_map(Salome2012())
+
+
+###########################################################
+#
+# Genome definition
+#
+###########################################################
+
+# Data for length information based on:
+# https://www.arabidopsis.org/portals/genAnnotation/
+#   gene_structural_annotation/agicomplete.jsp
+# Lengths from TAIR 10 although its unclear what reference the genetic map used
+#   -- follow up on this with Salome 2012 authors
+
+_chromosome_data = """\
+chr1 30427671
+chr2 19698289
+chr3 23459830
+chr4 18585056
+chr5 26975502
+"""
+
+_chromosomes = []
+for line in _chromosome_data.splitlines():
+    name, length = line.split()[:2]
+    _chromosomes.append(genomes.Chromosome(
+        name=name, length=int(length),
+        mean_mutation_rate=7e-9,  # rate from Ossowski 2010 Science
+        mean_recombination_rate=8.1e-9))  # value from Huber et al 2014 MBE
+                                          # rho=200/Mb,
+                                          # assume Ne=124,000, rho=2*Ne*r
+
+
+genome = genomes.Genome(
+    species="arabidopsis_thaliana",
+    chromosomes=_chromosomes,
+    default_genetic_map=Salome2012.name)
+
+
+###########################################################
+#
+# Demographic models
+#
+###########################################################
+
+
+class Durvasula2017MSMC(models.Model):
+    """
+    add info about model here
+
+    """
+
+    def __init__(self):
+       self.times = [699.1107142857144, 2796.4428571428575, 6067.585714285714,
```

Comment:
hey @jeromekelleher are you cool with this style for hard coded arrays? looks like this passes flake8 linting

Reply:
If it passes flake8 and it's not likely to *not* pass flake8 in the future I'm fine with it.

Reply:
The most recent version is a little cleaner: https://github.com/popgensims/stdpopsim/pull/23/files#diff-579791a7ad461a14fdba75fc383c1ae7R85